rskj-core/src/main/java/co/rsk/rpc/netty/Web3HttpStatusCodeProvider.java

```diff
@@ -0,0 +1,48 @@
+package co.rsk.rpc.netty;
+
+import com.googlecode.jsonrpc4j.HttpStatusCodeProvider;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.googlecode.jsonrpc4j.ErrorResolver.JsonError;
+import static com.googlecode.jsonrpc4j.ErrorResolver.JsonError.*;
+
+public class Web3HttpStatusCodeProvider implements HttpStatusCodeProvider {
+
+    private static final List<JsonError> BAD_REQUEST_JSON_ERRORS = Arrays.asList(
+            METHOD_PARAMS_INVALID, INVALID_REQUEST, PARSE_ERROR
+    );
+
+    /**
+     * Only invalid JSON-RPC requests (e.g. a malformed JSON) qualify for a Bad Request HTTP status.
+     * In any other case, the HTTP protocol remains entirely independent from JSON-RPC (since its 2.0 version).
+     * Reference: https://www.jsonrpc.org/
+     */
+    @Override
+    public int getHttpStatusCode(int resultCode) {
+        return isBadRequestResultCode(resultCode) ?
+                HttpResponseStatus.BAD_REQUEST.code() :
+                HttpResponseStatus.OK.code();
+    }
+
+    private boolean isBadRequestResultCode(int resultCode) {
+        Optional<JsonError> optionalJsonError = BAD_REQUEST_JSON_ERRORS
+                .stream()
+                .filter(jsonError -> jsonError.code == resultCode)
+                .findFirst();
+        return optionalJsonError.isPresent();
+    }
+
+    /**
+     * The default implementation wrongly assumes that an HTTP status code is mapped to a single JSON RPC error - this
+     * is not always the case. Either way, this is not used in the current project.
+     */
+    @Override
+    public Integer getJsonRpcCode(int httpStatusCode) {
+        throw new UnsupportedOperationException("Cannot possibly determine a single JSON RPC from an HTTP status code");
```

Comment:
jsonrpc4j maps each supported HTTP status code to a single JSON-RPC code - this is not entirely correct in my opinion, since a single HTTP code could map to N JSON-RPC codes (this even happens in jsonrpc4j).
Since we don't use this method in rskj, I decided to leave it like this.

Reply:
Take into account that a JSON RPC request could be an array of valid JSON RPC requests. Is this case covered in the process of responses/responses code?

Reply:
Since I just swapped the `DefaultHttpStatusCodeProvider` (from jsonrpc4j library) for a custom implementation that adheres to the same signatures (but mapping HTTP codes differently), the actual processing of requests is not involved (this is only used in the response handling part of the pipeline).